Cannot install on Ubuntu 22.04 (GitHub runner)

I get the following error: `xml_document.c:495:14: error: conflicting types for ‘canonicalize’; have`

<details><summary>Complete stacktrace</summary>
<pre><code>
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/nokogiri-1.10.10/ext/nokogiri
/opt/hostedtoolcache/Ruby/3.2.2/x64/bin/ruby extconf.rb --use-system-libraries
checking if the C compiler accepts ... no
Building nokogiri using system libraries.
checking for xmlParseDoc() in libxml/parser.h... yes
checking for xsltParseStylesheetDoc() in libxslt/xslt.h... yes
checking for exsltFuncRegister() in libexslt/exslt.h... yes
checking for xmlHasFeature()... yes
checking for xmlFirstElementChild()... yes
checking for xmlRelaxNGSetParserStructuredErrors()... yes
checking for xmlRelaxNGSetParserStructuredErrors()... yes
checking for xmlRelaxNGSetValidStructuredErrors()... yes
checking for xmlSchemaSetValidStructuredErrors()... yes
checking for xmlSchemaSetParserStructuredErrors()... yes
creating Makefile

current directory:
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/nokogiri-1.10.10/ext/nokogiri
make DESTDIR\= sitearchdir\=./.gem.20230510-1797-a6r78n
sitelibdir\=./.gem.20230510-1797-a6r78n clean

current directory:
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/nokogiri-1.10.10/ext/nokogiri
make DESTDIR\= sitearchdir\=./.gem.20230510-1797-a6r78n
sitelibdir\=./.gem.20230510-1797-a6r78n
compiling html_document.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./html_document.h:4,
                 from html_document.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:47,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./html_document.h:4,
                 from html_document.c:1:
html_document.c: In function ‘new’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
html_document.c:16:3: note: in expansion of macro ‘rb_scan_args’
   16 |   rb_scan_args(argc, argv, "0*", &rest);
      |   ^~~~~~~~~~~~
html_document.c: In function ‘init_html_document’:
html_document.c:152:6: warning: old-style function definition
[-Wold-style-definition]
  152 | void init_html_document()
      |      ^~~~~~~~~~~~~~~~~~
html_document.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling html_element_description.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./html_element_description.h:4,
                 from html_element_description.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
html_element_description.c: In function ‘init_html_element_description’:
html_element_description.c:256:6: warning: old-style function definition
[-Wold-style-definition]
  256 | void init_html_element_description()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
html_element_description.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling html_entity_lookup.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./html_entity_lookup.h:4,
                 from html_entity_lookup.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
html_entity_lookup.c: In function ‘init_html_entity_lookup’:
html_entity_lookup.c:25:6: warning: old-style function definition
[-Wold-style-definition]
   25 | void init_html_entity_lookup()
      |      ^~~~~~~~~~~~~~~~~~~~~~~
html_entity_lookup.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling html_sax_parser_context.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./html_sax_parser_context.h:4,
                 from html_sax_parser_context.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
html_sax_parser_context.c: In function ‘init_html_sax_parser_context’:
html_sax_parser_context.c:100:6: warning: old-style function definition
[-Wold-style-definition]
  100 | void init_html_sax_parser_context()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
html_sax_parser_context.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling html_sax_push_parser.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./html_sax_push_parser.h:4,
                 from html_sax_push_parser.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
html_sax_push_parser.c: In function ‘init_html_sax_push_parser’:
html_sax_push_parser.c:76:6: warning: old-style function definition
[-Wold-style-definition]
   76 | void init_html_sax_push_parser()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
html_sax_push_parser.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling nokogiri.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from nokogiri.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
nokogiri.c: In function ‘Init_nokogiri’:
nokogiri.c:62:6: warning: old-style function definition [-Wold-style-definition]
   62 | void Init_nokogiri()
      |      ^~~~~~~~~~~~~
nokogiri.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling xml_attr.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_attr.h:4,
                 from xml_attr.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:47,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_attr.h:4,
                 from xml_attr.c:1:
xml_attr.c: In function ‘new’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_attr.c:61:3: note: in expansion of macro ‘rb_scan_args’
   61 |   rb_scan_args(argc, argv, "2*", &document, &name, &rest);
      |   ^~~~~~~~~~~~
xml_attr.c: In function ‘init_xml_attr’:
xml_attr.c:88:6: warning: old-style function definition [-Wold-style-definition]
   88 | void init_xml_attr()
      |      ^~~~~~~~~~~~~
xml_attr.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling xml_attribute_decl.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_attribute_decl.h:4,
                 from xml_attribute_decl.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
xml_attribute_decl.c: In function ‘init_xml_attribute_decl’:
xml_attribute_decl.c:58:6: warning: old-style function definition
[-Wold-style-definition]
   58 | void init_xml_attribute_decl()
      |      ^~~~~~~~~~~~~~~~~~~~~~~
xml_attribute_decl.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling xml_cdata.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_cdata.h:4,
                 from xml_cdata.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:47,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_cdata.h:4,
                 from xml_cdata.c:1:
xml_cdata.c: In function ‘new’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_cdata.c:23:3: note: in expansion of macro ‘rb_scan_args’
   23 |   rb_scan_args(argc, argv, "2*", &doc, &content, &rest);
      |   ^~~~~~~~~~~~
xml_cdata.c: In function ‘init_xml_cdata’:
xml_cdata.c:45:6: warning: old-style function definition
[-Wold-style-definition]
   45 | void init_xml_cdata()
      |      ^~~~~~~~~~~~~~
xml_cdata.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling xml_comment.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_comment.h:4,
                 from xml_comment.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:47,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_comment.h:4,
                 from xml_comment.c:1:
xml_comment.c: In function ‘new’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_comment.c:21:3: note: in expansion of macro ‘rb_scan_args’
   21 |   rb_scan_args(argc, argv, "2*", &document, &content, &rest);
      |   ^~~~~~~~~~~~
xml_comment.c: In function ‘init_xml_comment’:
xml_comment.c:51:6: warning: old-style function definition
[-Wold-style-definition]
   51 | void init_xml_comment()
      |      ^~~~~~~~~~~~~~~~
xml_comment.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
compiling xml_document.c
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/assume.h:29,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/backward/2/assume.h:24,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/defines.h:72,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:25,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_document.h:4,
                 from xml_document.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:
In function ‘RARRAY_PTR’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/core/rarray.h:551:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  551 |     return RBIMPL_CAST((VALUE *)RARRAY_CONST_PTR(tmp));
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:
In function ‘rb_enc_step_back’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/encoding.h:749:24:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  749 |     return RBIMPL_CAST((char *)ur);
      |                        ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:
In function ‘rb_enc_is_newline’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:45:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   45 |     OnigUChar *up = RBIMPL_CAST((OnigUChar *)p);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/encoding/ctype.h:46:33:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   46 |     OnigUChar *ue = RBIMPL_CAST((OnigUChar *)e);
      |                                 ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/cast.h:31:29:
note: in definition of macro ‘RBIMPL_CAST’
   31 | # define RBIMPL_CAST(expr) (expr)
      |                             ^~~~
xml_document.c: In function ‘dealloc’:
xml_document.c:49:25: warning: passing argument 2 of ‘rb_st_foreach’ from
incompatible pointer type [-Wincompatible-pointer-types]
   49 |   st_foreach(node_hash, dealloc_node_i, (st_data_t)doc);
      |                         ^~~~~~~~~~~~~~
      |                         |
|                         int (*)(xmlNode *, xmlNode *, xmlDoc *) {aka int
(*)(struct _xmlNode *, struct _xmlNode *, struct _xmlDoc *)}
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/arithmetic/st_data_t.h:31,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/arithmetic.h:37,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:28,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_document.h:4,
                 from xml_document.c:1:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/st.h:141:31: note:
expected ‘int (*)(st_data_t,  st_data_t,  st_data_t)’ {aka ‘int (*)(long
unsigned int,  long unsigned int,  long unsigned int)’} but argument is of type
‘int (*)(xmlNode *, xmlNode *, xmlDoc *)’ {aka ‘int (*)(struct _xmlNode *,
struct _xmlNode *, struct _xmlDoc *)’}
  141 | int rb_st_foreach(st_table *, st_foreach_callback_func *, st_data_t);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:47,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_document.h:4,
                 from xml_document.c:1:
xml_document.c: In function ‘duplicate_document’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_document.c:330:6: note: in expansion of macro ‘rb_scan_args’
  330 |   if(rb_scan_args(argc, argv, "01", &level) == 0)
      |      ^~~~~~~~~~~~
xml_document.c: In function ‘new’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_document.c:357:3: note: in expansion of macro ‘rb_scan_args’
  357 |   rb_scan_args(argc, argv, "0*", &rest);
      |   ^~~~~~~~~~~~
xml_document.c: In function ‘create_entity’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_document.c:436:3: note: in expansion of macro ‘rb_scan_args’
  436 |   rb_scan_args(argc, argv, "14", &name, &type, &external_id, &system_id,
      |   ^~~~~~~~~~~~
xml_document.c: At top level:
xml_document.c:495:14: error: conflicting types for ‘canonicalize’; have
‘VALUE(int,  VALUE *, VALUE)’ {aka ‘long unsigned int(int,  long unsigned int *,
long unsigned int)’}
  495 | static VALUE canonicalize(int argc, VALUE* argv, VALUE self)
      |              ^~~~~~~~~~~~
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdlib.h:26,
                 from ./nokogiri.h:4,
                 from ./xml_document.h:4,
                 from xml_document.c:1:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:3[70](https://github.com/asciidoctor/asciidoctor-reveal.js/actions/runs/4939450156/jobs/8830211151#step:7:71):1: note: previous declaration
of ‘canonicalize’ with type ‘int(double *, const double *)’
370 | __MATHDECL_1 (int, canonicalize,, (_Mdouble_ *__cx, const _Mdouble_
*__x));
      | ^~~~~~~~~~~~
In file included from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/ruby.h:47,
from
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby.h:38,
                 from ./nokogiri.h:33,
                 from ./xml_document.h:4,
                 from xml_document.c:1:
xml_document.c: In function ‘canonicalize’:
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:375:62:
warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
375 |                      (rb_scan_args_verify(fmt, varc), vars), (char
*)fmt, varc)
      |                                                              ^
/opt/hostedtoolcache/Ruby/3.2.2/x64/include/ruby-3.2.0/ruby/internal/scan_args.h:498:9:
note: in expansion of macro ‘rb_scan_args0’
  498 |         rb_scan_args0(                                        \
      |         ^~~~~~~~~~~~~
xml_document.c:511:3: note: in expansion of macro ‘rb_scan_args’
  511 |   rb_scan_args(argc, argv, "03", &mode, &incl_ns, &with_comments);
      |   ^~~~~~~~~~~~
xml_document.c: In function ‘init_xml_document’:
xml_document.c:554:6: warning: old-style function definition
[-Wold-style-definition]
  554 | void init_xml_document()
      |      ^~~~~~~~~~~~~~~~~
xml_document.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been
intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have
been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may
have been intended to silence earlier diagnostics
make: *** [Makefile:248: xml_document.o] Error 1

make failed, exit code 2

Gem files will remain installed in
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/nokogiri-1.10.10
for inspection.
Results logged to
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/extensions/x86_64-linux/3.2.0/nokogiri-1.10.10/gem_make.out

/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:119:in
`run'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:51:in
`block in make'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:43:in
`each'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:43:in
`make'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/ext_conf_builder.rb:41:in
`build'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:187:in
`build_extension'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:221:in
`block in build_extensions'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:218:in
`each'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/ext/builder.rb:218:in
`build_extensions'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/installer.rb:843:in
`build_extensions'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/rubygems_gem_installer.rb:[72](https://github.com/asciidoctor/asciidoctor-reveal.js/actions/runs/4939450156/jobs/8830211151#step:7:73):in
`build_extensions'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/rubygems_gem_installer.rb:28:in
`install'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/source/rubygems.rb:200:in
`install'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/installer/gem_installer.rb:54:in
`install'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/installer/gem_installer.rb:16:in
`install_from_spec'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/installer/parallel_installer.rb:156:in
`do_install'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/installer/parallel_installer.rb:147:in
`block in worker_pool'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/worker.rb:62:in
`apply_func'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/worker.rb:57:in
`block in process_queue'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/worker.rb:54:in
`loop'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/worker.rb:54:in
`process_queue'
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/bundler/worker.rb:[90](https://github.com/asciidoctor/asciidoctor-reveal.js/actions/runs/4939450156/jobs/8830211151#step:7:91):in
`block (2 levels) in create_threads'

An error occurred while installing nokogiri (1.10.10), and Bundler cannot
continue.

In Gemfile:
  asciidoctor-doctest was resolved to 2.0.0.beta.7, which depends on
    nokogiri
</code></pre>
</details> 

Nokogiri fixed this bug in: https://github.com/sparklemotion/nokogiri/commit/b061bba3059e29b92921cf40bf729a0222150d30
Would it be possible to upgrade to Nokogiri 1.14 ? 

For reference, Asciidoctor is using:

```
s.add_development_dependency 'nokogiri', '~> 1.14.0'
```

https://github.com/asciidoctor/asciidoctor/blob/9f15a500f385ec2692a620e9e06203e56a59cba7/asciidoctor.gemspec#L43

Thanks!

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,10 +11,12 @@ jobs:
       fail-fast: false
       matrix:
         ruby:
+          - '3.4'
+          - '3.3'
+          - '3.2'
           - '3.1'
           - '3.0'
           - '2.7'
-          - '2.6'
           - head
     continue-on-error: ${{ endsWith(matrix.ruby, 'head') }}
     steps:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,7 +17,7 @@ jobs:
           - '3.1'
           - '3.0'
           - '2.7'
-          - head
+         #- head  TODO: Uncomment after asciidoctor fixes compatibility with Ruby >=2.5
     continue-on-error: ${{ endsWith(matrix.ruby, 'head') }}
     steps:
       - uses: actions/checkout@v3

--- a/asciidoctor-doctest.gemspec
+++ b/asciidoctor-doctest.gemspec
@@ -17,7 +17,7 @@ A tool for end-to-end testing of Asciidoctor backends based on comparing of text
   s.files         = Dir['data/**/*', 'lib/**/*', '*.gemspec', 'CHANGELOG*', 'LICENSE*', 'README*']
   s.require_paths = ['lib']
 
-  s.required_ruby_version = '>= 2.0'
+  s.required_ruby_version = '>= 2.7'
 
   # runtime
   s.add_runtime_dependency 'asciidoctor', '>= 1.5.0', '< 3.0'

--- a/asciidoctor-doctest.gemspec
+++ b/asciidoctor-doctest.gemspec
@@ -25,7 +25,7 @@ A tool for end-to-end testing of Asciidoctor backends based on comparing of text
   s.add_runtime_dependency 'diffy', '~> 3.0'
   s.add_runtime_dependency 'htmlbeautifier', '~> 1.0'
   s.add_runtime_dependency 'minitest', '~> 5.4'
-  s.add_runtime_dependency 'nokogiri', '~> 1.10.8'
+  s.add_runtime_dependency 'nokogiri', '~> 1.14.0'
 
   # development
   s.add_development_dependency 'bundler', '>= 1.6'

--- a/asciidoctor-doctest.gemspec
+++ b/asciidoctor-doctest.gemspec
@@ -24,7 +24,7 @@ A tool for end-to-end testing of Asciidoctor backends based on comparing of text
   s.add_runtime_dependency 'corefines', '~> 1.2'
   s.add_runtime_dependency 'diffy', '~> 3.0'
   s.add_runtime_dependency 'htmlbeautifier', '~> 1.0'
-  s.add_runtime_dependency 'minitest', '~> 5.4'
+  s.add_runtime_dependency 'minitest', '~> 5.25'
   s.add_runtime_dependency 'nokogiri', '~> 1.14.0'
 
   # development

--- a/asciidoctor-doctest.gemspec
+++ b/asciidoctor-doctest.gemspec
@@ -36,6 +36,7 @@ A tool for end-to-end testing of Asciidoctor backends based on comparing of text
   # unit tests
   s.add_development_dependency 'codeclimate-test-reporter', '~> 1.0'
   s.add_development_dependency 'fakefs', '~> 1.2'
+  s.add_development_dependency 'ostruct', '~> 0.6'
   s.add_development_dependency 'simplecov', '~> 0.17.1'
   s.add_development_dependency 'rspec', '~> 3.1'
   s.add_development_dependency 'rspec-collection_matchers', '~> 1.1'

--- a/lib/asciidoctor/doctest/html_normalizer.rb
+++ b/lib/asciidoctor/doctest/html_normalizer.rb
@@ -110,7 +110,7 @@ module Asciidoctor
 
       # @return [Boolean] true if the +node+ is descendant of +<pre>+ node.
       def preformatted_block?(node)
-        node.path =~ %r{/pre/}
+        node.ancestors('pre').any?
       end
     end
   end

--- a/lib/asciidoctor/doctest/tester.rb
+++ b/lib/asciidoctor/doctest/tester.rb
@@ -114,11 +114,9 @@ module Asciidoctor
 
         # @note Overrides method from +Minitest::Test+.
         def run
-          with_info_handler do
-            time_it do
-              capture_exceptions do
-                @callable.call(self)
-              end
+          time_it do
+            capture_exceptions do
+              @callable.call(self)
             end
           end
 

